server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -707,6 +763,16 @@ public static String getPinningEntity(String repositoryName, String snapshotUUID
         return repositoryName + SNAPSHOT_PINNED_TIMESTAMP_DELIMITER + snapshotUUID;
     }
 
+    public static String getPinningEntity(String blobname) {
```

Comment:
We can use `SnapshotsService.getPinningEntity`

Reply:
Keeping it as it is as we are using the cached values of pinned entities here .